redis_enterprise/datadog_checks/redis_enterprise/check.py

```diff
@@ -0,0 +1,117 @@
+from collections import ChainMap
+from typing import Any  # noqa: F401
+
+from datadog_checks.base import AgentCheck, ConfigurationError, OpenMetricsBaseCheckV2
+from datadog_checks.base.checks.openmetrics.v2.scraper import OpenMetricsCompatibilityScraper
+
+from .metrics import ADDITIONAL_METRICS, DEFAULT_METRICS
+
+
+class RedisEnterpriseCheck(OpenMetricsBaseCheckV2):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'rdse'
+
+    def __init__(self, name, init_config, instances):
+
+        super(RedisEnterpriseCheck, self).__init__(name, init_config, instances)
+        self.check_initializations.appendleft(self._parse_config)
+
+    def _parse_config(self):
+        self.scraper_configs = []
+        metrics_endpoint = self.instance.get('openmetrics_endpoint')
+        metrics = self.get_default_config()
+
+        additional = []
+        groups = self.instance.get('metric_groups', [])
+        for g in groups:
+            if g not in ADDITIONAL_METRICS:
+                raise ConfigurationError(f'invalid metric in config: {g}')
+            additional.append(ADDITIONAL_METRICS[g])
+
+        if len(additional) > 0:
+            self.service_check("more_groups", AgentCheck.OK)
+            metrics += additional
+
+        config = {
+            'openmetrics_endpoint': metrics_endpoint,
+            'namespace': self.__NAMESPACE__,
+            'metrics': metrics,
+            'metadata_label_map': {'version': 'version'},
+        }
+
+        config.update(self.instance)
+        self.scraper_configs.append(config)
+
+    def get_default_config(self):
+
+        metrics = []
+        for dm in DEFAULT_METRICS:
+            metrics.append(dm)
+        return metrics
+
+    def create_scraper(self, config):
+        return OpenMetricsCompatibilityScraper(self, self.get_config_with_defaults(config))
+
+    def get_config_with_defaults(self, config):
+        metrics = config.pop('metrics')
+        return ChainMap(config, {'metrics': metrics})
+
+    def can_connect(self, hostname=None, message=None, tags=None):
+        print(f'hostname: {hostname}, message: {message}, tags: {tags}')
+        return False
+
+    def check(self, instance):
+        # type: (Any) -> None
+
+        # smoke test
+        url = instance.get('openmetrics_endpoint')
+        if not url:
+            raise ConfigurationError('Configuration error, please fix conf.yaml')
+
+        try:
+            super().check(instance)
+            self.service_check("can_connect", AgentCheck.OK)
+
+        except Exception as e:
+            self.log.error('exception: %s', e)
+            self.service_check("can_connect", AgentCheck.CRITICAL, message=str(e))
+
+        # The following are useful bits of code to help new users get started.
```

Comment:
@j8-redis can provide insight on why this is included? Are users going to have to build on top of the integration? @iliakur in case you have insight on this

Reply:
it reports critical failure due to misconfiguration:
 requests.exceptions.ConnectionError: There was an error scraping endpoint https://localhost:8071/metrics: HTTPSConnectionPool(host='localhost', port=8071): Max retries exceeded with url: /metrics (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1067007d0>: Failed to establish a new connection: [Errno 61] Connection refused'))

without it 'test_emits_critical_service_check_when_service_is_down' will not pass

Reply:
the code created by the scaffolding has been removed